### ISSUE TITLE
Bug VBET 0.8.1 - Assembly of Pieces on First Level Path is not working correctly

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/riverscapes-tools#654]
From #652, we saw a problem that @philipbaileynar described as:

> Kelly is investigating why its missing the mainstem CL and IGOs.

@joewheaton described as:

> We have a problem with assembly around the first Level Path that in 90% of these HUCs look way worse than they actually are. It is often (not always) manifested in the bottom of the HUC looking off. Needs research and fix!

**Comment by joewheaton** [comment]
Lets reserve judgement on this one until #655 and #659 are better understood. I think those are the root causes. What @philipbaileynar  and @joewheaton revealed in a chat (copying here)

> This "assembly" problem. Are you referring to missing cl and iGO, or are you referring to bloating/enlargement of vb near outflow of each huc?
> 
> Joseph Wheaton, Yesterday 5:24 PM
> I don't know what the bloating/enlargement thing is.  I cannot see the missing CL or IGO thing yet as our business logic doesn't have that shit yet so it is excluded from WebRV. I will tackle it shortly in QRV. The valley bottoms on about 85% of the HUC10s are messed up for the longest level path. I suspect it is related....
> Actually... I just found on 0514020112 that it seems to have CL okay, so not sure.
> 
> Philip Bailey, Yesterday 5:33 PM
> When the mainstem cl is missing it's usually because the end points of the cl are microscopically off the end of the raster. Kelly already working on a fix.
> 
> Joseph Wheaton, Yesterday 5:34 PM
> This is something different. When you get a chance to watch the video, you will see it over and over. I will keep working to see if I can unpack what is going on.
> That fix will be helpful... this is more fundamental.

**Comment by joewheaton** [comment]
## `EMPTY_CHANNEL_AREA` bug illustrated at Kiowa Creek
Okay, pattern on is making sense on this one...   Watch this [shorter video https://youtu.be/7jROvQKNtf8](https://youtu.be/7jROvQKNtf8) for [HUC 1019001003 - Kiowa Creek](https://northarrowresearchlabs.github.io/webrave-staging/#/Anabranch/b396e0f2-225e-4b06-b4cb-9aebeffe8ef0).

**Comment by philipbaileynar** [comment]
- [VBET project](https://northarrowresearchlabs.github.io/webrave-staging/#/Anabranch/df004837-2859-464a-8d95-198fe014e485)
- [Corresponding channel area project](https://northarrowresearchlabs.github.io/riverscapes-staging//#/Anabranch/cd81392f-d0b7-4059-85b0-08d307992558)

We have found a channel area feature with `geometry.IsEmpty` is True. The attributes for this channel area project feature are:

```csv
wkt_geom,fid,Permanent_,FDate,Resolution,GNIS_ID,GNIS_Name,LengthKM,ReachCode,FlowDir,WBArea_Per,FType,FCode,MainPath,InNetwork,Visibility,Shape_Leng,NHDPlusID,VPUID,Enabled,AreaSqKm,TotDASqKm,DivDASqKm,Slope,MaxElevSmo,MInElevSmo,BFwidth,bankfull_m,Elevation,Shape_Area,ca_temp_area
MultiPolygon EMPTY,246,43987804,2012-01-21,2,01438655,Big Creek,0.044,16010201000242,1,,460,46006,0,1,0,0.000429481573085,70000600002549,1601,1,0.0071,4.67920005,4.67920005,1e-05,181345,181345,2.047810564818457,2.053954488646375,0,0,0
```

<img width="894" alt="Screenshot 2022-12-05 at 5 00 25 PM" src="https://user-images.githubusercontent.com/8645379/205779924-5594b706-933f-4bad-a4f3-cd6a81d255f8.png">

**Comment by MattReimer** [comment]
There's a new check now across all our tools that filters out zero-area features. Previously we were checking for null geometry (which is still a thing we're filtering out) but `area === 0` is a subtely different kind of problem we need to check for explicitly.

We already filtered zero-length lines so this is much more consistent behaviour for us. Anyway. These features should no longer appear in any of our tools going foreard.

**Comment by philipbaileynar** [comment]
@joewheaton Matt's fix above is part 1 of two... It will prevent our "feature class copy code" from propagating erroneous features. For obvious reasons, at the first sign of missing features users should head straight to the log files for the upstream tools where the skipping of empty geometries will be logged.

Part 2...  @KellyMWhitehead is also researching why channel area features are getting produced with zero area. There's probably a case in our implementation of Beechie Imaki implementation that's awry. Perhaps NULL drainage area screws up a fraction or power function.

**Comment by MattReimer** [comment]
Re: Part 2: 

the RSContext project for this huc: `1601020101`  is fine without any 0-area shapes but...

The 0-area shape is coming from Channel Area

I have added the same filters to `merge_feature_classes` that we dod on `copy_feature_classes` and now channel area no longer has that 0-area shape.

**Comment by MattReimer** [comment]
Fixed as of 0.8.3
